worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java

```diff
@@ -117,36 +117,32 @@ public static <B extends BlockStateHolder<B>> B transform(B block, Transform tra
         checkNotNull(block);
         checkNotNull(transform);
 
+        if (transform.isIdentity()) {
+            return block;
+        }
+
         B result = block;
         List<? extends Property<?>> properties = block.getBlockType().getProperties();
 
         for (Property<?> property : properties) {
-            if (property instanceof DirectionalProperty) {
-                DirectionalProperty dirProp = (DirectionalProperty) property;
+            if (property instanceof DirectionalProperty dirProp) {
                 Direction value = (Direction) block.getState(property);
                 if (value != null) {
                     Vector3 newValue = getNewStateValue(dirProp.getValues(), transform, value.toVector());
                     if (newValue != null) {
                         result = result.with(dirProp, Direction.findClosest(newValue, Direction.Flag.ALL));
                     }
                 }
-            } else if (property instanceof EnumProperty) {
-                EnumProperty enumProp = (EnumProperty) property;
+            } else if (property instanceof EnumProperty enumProp) {
                 if (property.getName().equals("axis")) {
                     // We have an axis - this is something we can do the rotations to :sunglasses:
                     Direction value = null;
                     switch ((String) block.getState(property)) {
-                        case "x":
-                            value = Direction.EAST;
-                            break;
-                        case "y":
-                            value = Direction.UP;
-                            break;
-                        case "z":
-                            value = Direction.NORTH;
-                            break;
-                        default:
-                            break;
+                        case "x" -> value = Direction.EAST;
+                        case "y" -> value = Direction.UP;
+                        case "z" -> value = Direction.NORTH;
+                        default -> {
+                        }
```

Comment:
Should be switch expression, e. g.
```java
var value = switch(...) {...}
```
`null` can be returned from the default branch.